src/core/qgstemporalnavigationobject.cpp

```diff
@@ -264,7 +277,7 @@ void QgsTemporalNavigationObject::skipToEnd()
 long long QgsTemporalNavigationObject::totalFrameCount()
 {
   QgsInterval totalAnimationLength = mTemporalExtents.end() - mTemporalExtents.begin();
-  return std::floor( totalAnimationLength.seconds() / mFrameDuration.seconds() ) + 1;
+  return std::floor( totalAnimationLength.seconds() / QgsInterval( mFrameTimeStep, mFrameTimeStepUnit ).seconds() ) + 1;
```

Comment:
Because this calculation differs from the actual results obtained when stepping through frames, it's possible that the count returned here may differ from the actual number of frames. Can you update this calculation to exactly match the number of frames using the new time step calculation?